pydicom/color.py

```diff
@@ -0,0 +1,149 @@
+    have_pillow = False
+
+
+if have_numpy and have_pillow:
+
+    class ColorManager(object):
+
+        """Class for color management using ICC profiles."""
+
+        def __init__(self, icc_profile: bytes):
+            """
+
+            Parameters
+            ----------
+            icc_profile: bytes
+                ICC profile
+
+            Raises
+            ------
+            ValueError
+                When ICC Profile cannot be read.
+            RuntimeError
+                When color management is not possible due to missing
+                dependencies.
+
+            """
+            try:
+                self._icc_transform = self._build_icc_transform(icc_profile)
+            except OSError:
+                raise ValueError('Could not read ICC Profile.')
+
+        def transform_frame(self, array: np.ndarray) -> np.ndarray:
+            """Transforms a frame by applying the ICC profile.
+
+            Parameters
+            ----------
+            array: numpy.ndarray
+                Pixel data of a color image frame in form of an array with
+                dimensions (Rows x Columns x SamplesPerPixel)
+
+            Returns
+            -------
+            numpy.ndarray
+                Color corrected pixel data of a image frame in form of an array
+                with dimensions (Rows x Columns x SamplesPerPixel)
+
+            Raises
+            ------
+            ValueError
+                When `array` does not have 3 dimensions and thus does not represent
+                a color image frame.
+
+            """
+            if array.ndim != 3:
+                raise ValueError(
+                    'Array has incorrect dimensions for a color image frame.'
+                )
+            image = Image.fromarray(array)
+            applyTransform(image, self._icc_transform, inPlace=True)
+            return np.asarray(image)
+
+        @staticmethod
+        def _build_icc_transform(icc_profile: bytes) -> ImageCmsTransform:
+            """Builds an ICC Transformation object.
+
+            Parameters
+            ----------
+            icc_profile: bytes
+                ICC Profile
+
+            Returns
+            -------
+            PIL.ImageCms.ImageCmsTransform
+                ICC Transformation object
+
+            """
+            profile: bytes
+            try:
+                profile = ImageCmsProfile(BytesIO(icc_profile))
+            except OSError:
+                raise ValueError('Cannot read ICC Profile in image metadata.')
+            name = getProfileName(profile).strip()
+            description = getProfileDescription(profile).strip()
+            logger.debug(f'found ICC Profile "{name}": "{description}"')
+
+            logger.debug('build ICC Transform')
+            intent = ImageCms.INTENT_RELATIVE_COLORIMETRIC
+            if not isIntentSupported(
+                profile,
+                intent=intent,
+                direction=ImageCms.DIRECTION_INPUT
+            ):
```

Comment:
I know almost nothing about ICC profiles, but is relative colorimetric correct? [Standard has](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.15.html):

> The following constraints on the encoding of the ICC Profile are recommended:
> The Rendering Intent should be Perceptual.

And that's only a recommendation, not a requirement.

Reply:
Also not sure what intent would be best suited for the library. We may want to add a parameter so that the caller can change the behavior if necessary. Thoughts?

Reply:
@dclunie do you have any thoughts on this?